dali/operators/generic/slice/slice_attr.h

```diff
@@ -246,6 +239,26 @@ class PositionalSliceAttr {
   }
 
  private:
+  template <typename T>
+  span<T> Span(const TensorListView<StorageCPU, T, DynamicDimensions>& v, int sample_idx) {
+    ptrdiff_t vol = volume(v.tensor_shape_span(sample_idx));
+    return span<T>(v.tensor_data(sample_idx), vol);
+  }
+
+  template <typename T, typename Backend>
+  const TensorListView<StorageCPU, T, DynamicDimensions> GetInputArgumentCPU(
+      const workspace_t<Backend>& ws, int idx, TensorList<CPUBackend>& cpu_buffer) {
+    if (ws.template InputIsType<CPUBackend>(idx)) {
+      const auto &arg = ws.template Input<CPUBackend>(idx);
+      return view<T>(arg);
+    } else {
+      const auto &arg = ws.template Input<GPUBackend>(idx);
+      cpu_buffer.set_order(AccessOrder::host());
+      cpu_buffer.Copy(arg);
+      return view<T>(cpu_buffer);
```

Comment:
I have mixed feeling about this. We ask the caller to provide `cpu_buffer` only to returns its view later.
Maybe we can allocate the memory here dynamically and make the view sure that the underlining memory is alive as long as the view is alived(just a thought)?

Reply:
This function doesn't necessarily return the view to this buffer (if the argument is already on host). We need to manage the lifetime of the storage somehow and I think that, given the expected small size, we can afford to keep it as a member.